Formula/firefoxpwa.rb

```diff
@@ -0,0 +1,83 @@
+class Firefoxpwa < Formula
+  desc "Tool to install, manage and use Progressive Web Apps in Mozilla Firefox"
+  homepage "https://github.com/filips123/FirefoxPWA"
+  url "https://github.com/filips123/FirefoxPWA/archive/refs/tags/v0.5.0.tar.gz"
+  sha256 "40004c7a268436173813024672ea23599e387766973d36d0ee0b81793a20b169"
+  license "MPL-2.0"
+  head "https://github.com/filips123/FirefoxPWA.git"
+
+  depends_on "rust" => :build
+
+  def install
+    cd "native"
+
+    # Set the correct version in the source files
+    inreplace "Cargo.toml",
+              "version = \"0.0.0\"",
+              "version = \"#{version}\""
+    inreplace "userchrome/profile/chrome/pwa/chrome.jsm",
+              "DISTRIBUTION_VERSION = '0.0.0'",
+              "DISTRIBUTION_VERSION = '#{version}'"
+
+    # Set the path in the manifest to the Homebrew libexec directory
+    # This is needed so the program respects Homebrew's installation hierarchy
+    # We can use macOS manifest for Linux because the only difference is a default path
+    inreplace "manifests/macos.json",
+              "/usr/local/libexec/firefoxpwa-connector",
+              "#{libexec}/firefoxpwa-connector"
+
+    # Set the system data path to the Homebrew share directory
+    # This is needed so the program respects Homebrew's installation hierarchy
+    ENV["FFPWA_SYSDATA"] = share
+
+    # Build the native part
+    # We cannot use "cargo install" to automatically install it
+    # Because the package contains multiple binary and other files
+    system "cargo", "build", "--release"
+
+    # Install all files
+    bin.install "target/release/firefoxpwa"
+    libexec.install "target/release/firefoxpwa-connector"
+    share.install "manifests/macos.json" => "firefoxpwa.json"
+    share.install "userchrome/"
+    bash_completion.install "target/release/completions/firefoxpwa.bash" => "firefoxpwa"
+    fish_completion.install "target/release/completions/firefoxpwa.fish"
+    zsh_completion.install "target/release/completions/_firefoxpwa"
+  end
+
+  def post_install
+    ohai "You have successfully installed the native part of the FirefoxPWA project"
+    ohai "You should also install the Firefox extension if you haven't already"
+    ohai "Download: https://addons.mozilla.org/firefox/addon/firefoxpwa/"
+  end
+
+  def caveats
+    filename = "firefoxpwa.json"
+
+    source = share
+    destination = ""
+
+    on_macos do
+      destination = "/Library/Application Support/Mozilla/NativeMessagingHosts"
+    end
+
+    on_linux do
+      destination = "/usr/lib/mozilla/native-messaging-hosts"
+    end
+
+    <<~EOS
+      Before you can use the browser extension, you will need to manually link the app manifest
+      This is needed because Homebrew formulae cannot access directories outside Homebrew directory
+
+      #{Tty.magenta}#{Tty.bold}To link the manifest, run the following commands after the formula is installed:#{Tty.reset}
```

Comment:
Please don't use colors when they're not needed.

Reply:
I kept color for the final warning because it might be good that it's more noticeable, but I can also remove it if needed.